channels/factories.py

```diff
@@ -152,27 +157,47 @@ def get_instances(self, factory_type):
 
         return self.data[factory_type]
 
-    def get_or_create(self, factory_cls, factory_type, ident, **kwargs):
+    def build_or_create(self, factory_cls, strategy, kwargs):
         """
-        Creates a new named post from the provided factory
+        Creates a new named object from the provided factory
+
+        Args:
+            factory_cls (cls): class of the factory to generate the post
+            strategy (str): either a create or build strategy for the factory
+            kwargs (dict): the kwargs to pass to the factory
+
+        Returns:
+            object: the created object
+        """
+        if strategy == STRATEGY_CREATE:
+            return factory_cls.create(**kwargs)
+        elif strategy == STRATEGY_BUILD:
+            return factory_cls.build(**kwargs)
```

Comment:
what's the difference between `.create` and `.build` here?

Reply:
`factoryboy` explains the difference here: http://factoryboy.readthedocs.io/en/latest/reference.html#strategies

TLDR: `create` means persist, `build` means don't

Reply:
Ok I figured that was what was going on, sounds good.